react_juce/core/TextInputView.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+==============================================================================
+
+  TextInputView.cpp
+  Created: 20 Jan 2021 10:30pm
+
+==============================================================================
+*/
+
+#include "TextInputView.h"
+#include "TextView.h"
+
+namespace blueprint
+{
+    namespace detail
+    {
+        static juce::var makeInputEventObject(const juce::String &value)
+        {
+            juce::DynamicObject::Ptr obj = new juce::DynamicObject();
+            obj->setProperty("value", value);
+            return obj.get();
+        }
+
+        static juce::var makeChangeEventObject(const juce::String &value)
+        {
+            juce::DynamicObject::Ptr obj = new juce::DynamicObject();
+            obj->setProperty("value", value);
+            return obj.get();
+        }
+    }
+
+    void TextInput::setControlledValue(const juce::String &value)
+    {
+        insertedAsControlledValue = true;
+        setText(value);
+    }
+
+    void TextInput::setMaxLength(int maxLen)
+    {
+        maxLength = maxLen;
+        setInputRestrictions(maxLen);
+    }
+
+    void TextInput::setPlaceholderText(const juce::String &text)
+    {
+        placeholderText = text;
+        setTextToShowWhenEmpty(placeholderText, placeholderColour);
+    }
+
+    void TextInput::setPlaceholderColour(const juce::Colour &colourToUse)
+    {
+        placeholderColour = colourToUse;
+        setTextToShowWhenEmpty(placeholderText, placeholderColour);
+    }
+
+    void TextInput::insertTextAtCaret(const juce::String &textToInsert)
+    {
+        juce::String currentValue = getText();
```

Comment:
`currentValue` and `newValue` could be `const`

Reply:
I fixed it in 973a63c.